README.md

```diff
@@ -13,3 +13,139 @@ This is an NWB extension for storing metadata of devices used in optical experim
+        numerical_aperture : numeric, optional
+        core_diameter_in_um : numeric, optional
+    }
+    class ExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        illumination_type : text, optional
+        excitation_wavelength_in_nm : numeric, optional
+        power_in_W : numeric, optional
+        intensity_in_W_per_m2 : numeric, optional
+        exposure_time_in_s : numeric, optional
+    }
+    class PulsedExcitationSource{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        peak_power_in_W : numeric, optional
+        peak_pulse_energy_in_J : numeric, optional
+        pulse_rate_in_Hz : numeric, optional
+    }
+    class Photodetector{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        detector_type : text, optional
+        detected_wavelength_in_nm : numeric, optional
+        gain : numeric, optional
+    }
+    class DichroicMirror{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_on_wavelength_in_nm : numeric, optional
+        cut_off_wavelength_in_nm : numeric, optional
+        reflection_bandwidth_in_nm : numeric, optional
+        transmission_bandwidth_in_nm : numeric, optional
+        angle_of_incidence_in_degrees : numeric, optional
+    }
+    class OpticalFilter{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        filter_type : text, optional
+    }
+    class BandOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        center_wavelength_in_nm : numeric
+        bandwidth_in_nm : numeric
+    }
+    class EdgeOpticalFilter{
+        <<OpticalFilter>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        cut_wavelength_in_nm : numeric
+        slope_in_percent_cut_wavelength : numeric, optional
+        slope_starting_transmission_in_percent : numeric, optional
+        slope_ending_transmission_in_percent : numeric, optional
+    }
+    class ObjectiveLens{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        numerical_aperture : numeric, optional
+        magnification : numeric, optional
+    }
+    class Microscope{
+        <<DeviceModel>>
+        --------------------------------------
+        attributes
+        --------------------------------------
+        microscopy_type : text, optional
+        doi : text, optional
```

Comment:
All microscopes have a DOI? I don't think I've ever seen that in the methods/supplemental section

How would I look it up for a particular microscope?

Reply:
It was meant to be there just for custom microscopes. I remember we discussed this a while ago, but I am also not totally convinced that this is something needed.

Reply:
Where are we at on this? Does the base PyNWB device has the `manufacturer` then you add the `model` to the new `BaseModel`?